server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClient.java

```diff
@@ -69,4 +69,9 @@ public interface CoordinatorClient
    * Returns a new instance backed by a ServiceClient which follows the provided retryPolicy
    */
   CoordinatorClient withRetryPolicy(ServiceRetryPolicy retryPolicy);
+
+  /**
+   * Retrieves list of used datasources.
+   */
+  ListenableFuture<Set<String>> fetchUsedDataSources();
```

Comment:
I have updated the method name to `fetchDatasourcesWithUsedSegments` to make it more understandable. 
I don't think we need to document about what `used segments` means, since it is widely referred in the code and docs. For example, here https://druid.apache.org/docs/latest/api-reference/data-management-api/#mark-a-single-segment-as-used.